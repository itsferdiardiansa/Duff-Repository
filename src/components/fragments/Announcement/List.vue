<template>
  <div class="container">
    <div class="wrapper">
      <Table
        :headers="tHeaders"
        :items="filteredData"
        :showLoader="isFetching"
        :rowLoader="2"
        :selectableRows="true"
        :pagination="paginationData"
        @onSelectedRowCallback="handleSelectRow"
      >
        <template #link="props">
          <a :href="props.data.link" target="_blank" class="underline"
            >Selengkapnya</a
          >
        </template>

        <template #status="props">
          <Status :data="props.data.status" />
        </template>

        <template #action="props">
          <ActionButton :data="props.data" />
        </template>
      </Table>
    </div>
  </div>
</template>
<script>
import { computed, onMounted, ref } from 'vue'
import { useStore } from 'vuex'
import Table from '@common/Table'
import ActionButton from './ActionButton'
import Status from './Status'

export default {
  components: {
    Table,
    ActionButton,
    Status,
  },
  setup() {
    const store = useStore()

    let tHeaders = ref([
      {
        title: 'Title',
        accessor: 'title',
        width: '30%',
      },
      { title: 'Content', accessor: 'content' },
      { title: 'Link', accessor: 'link', align: 'center' },
      { title: 'Status', accessor: 'status', width: '9%', align: 'center' },
      { title: 'Expired', accessor: 'expired', align: 'center' },
      {
        title: 'Action',
        accessor: 'action',
        colSpan: 3,
        align: 'center',
      },
    ])

    const getAnnouncement = () => {
      store.dispatch('announcement/fetchData')
    }

    const filteredData = computed(() => (
      store.getters['announcement/getAnnouncement']
    ))

    const paginationData = computed(() => (
      store.getters['announcement/getPagination']
    ))

    const isFetching = computed(() => (
      store.getters['announcement/getFetchStatus']
    ))

    const searchFilter = data => {
      console.log(data, 'List')
    }

    const handleSelectRow = () => {
      // console.log(data)
    }

    onMounted(() => {
      getAnnouncement()
    })

    return {
      isFetching,
      filteredData,
      paginationData,
      tHeaders,
      searchFilter,
      handleSelectRow
    }
  },
}
</script>
<style lang="scss" scoped>
.container {
  @apply relative;

  .wrapper {
    @apply relative;
  }
}
</style>
