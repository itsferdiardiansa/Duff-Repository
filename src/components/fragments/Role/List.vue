<template>
  <div class="list">
    <Modal
      title="Delete confirmation"
      description="Are you sure you want to delete this item?"
      :onConfirmFn="deleteData"
    />

    <Table
      :headers="tHeaders"
      :items="filteredData"
      :isFetching="requestStatus.fetch"
      :onError="requestStatus.error.status"
      :onFailedFetchHandler="fetchData"
      :pagination="pagination"
      :onPageChange="handlePageChange"
    >
      <template #previlage="{ data }">
        <div class="previlage-col">
          <template v-for="(item, key) in data.previlege" :key="key">
            <Badge variant="primary" :inverse="true" class="mr-2">
              {{ item.id_privilege }}
            </Badge>
          </template>
        </div>
      </template>

      <template #action="{ data }">
        <ActionButton :data="actionButtons" :item="data" />
      </template>
    </Table>
  </div>
</template>
<script>
import { computed, onMounted, reactive, ref } from 'vue';
import { useStore } from 'vuex';
import Table, {
  EmptyTable as emptyComponent,
  ActionButton,
} from '@common/Table';
import Badge from '@common/Badge';
import Modal from '@common/Modal';
import { useRouter } from 'vue-router';

export default {
  components: {
    Table,
    Badge,
    Modal,
    ActionButton,
  },
  setup() {
    const store = useStore();
    const router = useRouter();
    const params = reactive({ page: 1, limit: 5 });

    const tHeaders = ref([
      {
        title: 'Role Name',
        accessor: 'name',
        width: '20%',
      },
      { title: 'Previlage', accessor: 'previlage' },
      { title: 'Created at', accessor: 'created_at' },
      { title: 'Updated at', accessor: 'updated_at' },
      {
        title: 'Action',
        accessor: 'action',
        colSpan: 1,
        align: 'center',
      },
    ]);

    const actionButtons = ref([
      {
        icon: ['fa', 'pencil-alt'],
        variant: 'dark',
        onClickFn: (e, data) => {
          router.push({
            name: 'Update Role',
            params: { data: JSON.stringify(data) },
          });
        },
      },
      {
        icon: ['fa', 'trash'],
        variant: 'dark',
        onClickFn: (e, data) => {
          toggleModal(e, data);
        },
      },
    ]);

    const toggleModal = (e, data) => {
      self.$modal.show(data);
    };

    const deleteData = ({ hash_id }) => {
      store.dispatch('role/deleteData', {
        action: 'form.delete',
        hash_id,
        params,
      });
    };

    const fetchData = () => {
      store.dispatch('role/fetchData', params);
    };

    const filteredData = computed(() => {
      return store.getters['role/getItems'];
    });

    const pagination = computed(() => {
      return store.getters['role/getPagination'];
    });

    const requestStatus = computed(() => {
      return store.getters['role/getRequestStatus'];
    });

    const handlePageChange = pageParams => {
      Object.assign(params, pageParams);

      fetchData();
    };

    onMounted(fetchData);

    return {
      requestStatus,
      filteredData,
      tHeaders,
      fetchData,
      deleteData,
      emptyComponent,
      actionButtons,
      toggleModal,
      pagination,
      handlePageChange,
    };
  },
};
</script>
<style lang="scss" scoped>
.content-container {
  @apply relative;

  .wrapper {
    @apply relative;
  }

  .previlage-col {
    @apply flex justify-center;
  }
}
</style>
