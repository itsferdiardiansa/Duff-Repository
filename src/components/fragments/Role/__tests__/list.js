import { mount } from '@vue/test-utils';
import { createStore } from 'vuex';
import List from '@fragment/Role/List';
import roleModule from '@store/modules/role';

describe('fragment/Role/List', () => {
  let store;

  beforeAll(() => {
    store = createStore({
      modules: {
        role: roleModule,
      },
    });
  });

  it('will render and match snapshot', async () => {
    const Component = mount(<List />, {
      global: {
        plugins: [store],
      },
    });

    await Component.vm.$store.dispatch('role/fetchData', {
      page: 1,
      limit: 10,
    });
    expect(Component.html()).toMatchSnapshot();
  });
});
