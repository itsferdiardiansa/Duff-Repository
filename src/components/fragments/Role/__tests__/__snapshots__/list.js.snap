// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragment/Role/List will render and match snapshot 1`] = `
<div class="list">
  <!--v-if-->
  <!--v-if-->
  <table class="ss-table">
    <thead class="ss-table--head" headers="[object Object],[object Object],[object Object],[object Object],[object Object]" items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" isfetching="false" pagination="[object Object]" emptymessage="Empty data" rownumber="true" rowloader="5" selectablerows="false" withpagination="true" withfilter="false" filteroptions="">
      <tr>
        <th width="4%">
          <!--v-if--><span>#</span>
        </th>
        <th class="" width="20%">Role Name</th>
        <th class="">Previlage</th>
        <th class="">Created at</th>
        <th class="">Updated at</th>
        <th colspan="1" class="" align="center">Action</th>
      </tr>
    </thead>
    <tbody class="ss-table--body">
      <!--v-if-->
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>1</label></td>
        <td class="ss-table--body-col">
          <div>dav create an role has been update</div>
        </td>
        <td class="ss-table--body-col">
          <div class="previlage-col"></div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-26T06:51:08Z</div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-26T08:06:46Z</div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183642">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183642">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 448 512">
                      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>2</label></td>
        <td class="ss-table--body-col">
          <div>new role updated</div>
        </td>
        <td class="ss-table--body-col">
          <div class="previlage-col"></div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-16T05:15:58Z</div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-03-04T04:22:43Z</div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183648">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183648">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 448 512">
                      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>3</label></td>
        <td class="ss-table--body-col">
          <div>new name</div>
        </td>
        <td class="ss-table--body-col">
          <div class="previlage-col"></div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-16T05:02:58Z</div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-16T05:02:58Z</div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183654">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183654">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 448 512">
                      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>4</label></td>
        <td class="ss-table--body-col">
          <div>updated yess</div>
        </td>
        <td class="ss-table--body-col">
          <div class="previlage-col"></div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-15T11:03:01Z</div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-16T05:19:10Z</div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183659">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183659">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 448 512">
                      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>5</label></td>
        <td class="ss-table--body-col">
          <div></div>
        </td>
        <td class="ss-table--body-col">
          <div class="previlage-col"></div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-15T10:56:18Z</div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-15T10:56:18Z</div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183670">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183670">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 448 512">
                      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>6</label></td>
        <td class="ss-table--body-col">
          <div>Make new role</div>
        </td>
        <td class="ss-table--body-col">
          <div class="previlage-col"></div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-10T08:07:44Z</div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-10T08:07:44Z</div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183675">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183675">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 448 512">
                      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>7</label></td>
        <td class="ss-table--body-col">
          <div>New top</div>
        </td>
        <td class="ss-table--body-col">
          <div class="previlage-col"></div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-10T05:37:11Z</div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-10T05:37:11Z</div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183687">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183687">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 448 512">
                      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>8</label></td>
        <td class="ss-table--body-col">
          <div>New create</div>
        </td>
        <td class="ss-table--body-col">
          <div class="previlage-col"></div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-10T05:35:56Z</div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-10T05:35:56Z</div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183694">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183694">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 448 512">
                      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>9</label></td>
        <td class="ss-table--body-col">
          <div>Anyone</div>
        </td>
        <td class="ss-table--body-col">
          <div class="previlage-col"></div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-10T04:41:04Z</div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-10T04:41:04Z</div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183702">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183702">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 448 512">
                      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>10</label></td>
        <td class="ss-table--body-col">
          <div>Error everywhere</div>
        </td>
        <td class="ss-table--body-col">
          <div class="previlage-col"></div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-10T04:40:34Z</div>
        </td>
        <td class="ss-table--body-col">
          <div>2021-02-10T04:40:34Z</div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183708">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button><button class="ss-button ss-button--dark ss-button--base" data-id="1614833183708">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 448 512">
                      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"></path>
                    </svg></div>
                  <!--v-if-->
                </div>
              </button></div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <div class="ss-pagination">
    <div class="ss-pagination--info">
      <p> Showing <span>1</span> to <span>10</span> of <span>20</span> results </p>
    </div>
    <div class="ss-pagination--page">
      <!--v-if--><span class="page-number active">1</span><span class="page-number">2</span><span class="nav-btn next"><img svg-inline=""></span>
    </div>
  </div>
</div>
`;
